src/pages/ViewProfilePage.jsx

```diff
@@ -1,12 +1,66 @@
 import React from 'react';
+import { connect } from 'react-redux';
+import firebase from 'firebase/app';
 import PageTitle from '../atoms/PageTitle';
+import user from '../Redux/reducers/user';
+import TextInputRow from '../atoms/TextInputRow';
+import SubmitButton from '../atoms/SubmitButton';
+import UserChange from '../Redux/actions/UserChange';
+import store from '../Redux/store';
 
 function ViewProfilePage() {
+  const [userInfo, setUserInfo] = React.useState({
+    Name: store.getState().user.Name, Email: store.getState().user.Email, Password: '', Message: '',
+  });
+
+  function handleNameInput(input) { setUserInfo({ ...userInfo, Name: input }); }
+  function handleEmailInput(input) { setUserInfo({ ...userInfo, Email: input }); }
+  function handlePasswordInput(input) { setUserInfo({ ...userInfo, Password: input }); }
+
+  async function newNameSubmission() {
+    await firebase.auth().currentUser.updateProfile({ displayName: userInfo.Name }).then(() => {
+      store.dispatch(UserChange({ ...store.getState().user, Name: userInfo.Name }));
+      setUserInfo({ ...userInfo, Message: 'Name successfully changed' });
+    }).catch((error) => {
+      setUserInfo({ ...userInfo, Message: error.message });
+    });
+  }
+
+  async function newEmailSubmission() {
+    await firebase.auth().currentUser.updateEmail(userInfo.Email).then(() => {
+      store.dispatch(UserChange({ ...store.getState().user, Email: userInfo.Email }));
+      setUserInfo({ ...userInfo, Message: 'Email successfully changed' });
+    }).catch((error) => {
+      setUserInfo({ ...userInfo, Message: error.message });
+    });
+  }
+
+  async function newPasswordSubmission() {
+    await firebase.auth().currentUser.updatePassword(userInfo.Password).then(() => {
+      setUserInfo({ ...userInfo, Message: 'Password successfully changed' });
+    }).catch((error) => {
+      setUserInfo({ ...userInfo, Message: error.message });
+    });
+  }
+
   return (
     <div data-test="ViewProfilePage" className="container-fluid p-padding text-center">
       <PageTitle Title="View Profile Page" />
+      <form data-test="submitName" onSubmit={newNameSubmission}>
```

Comment:
Should probably just have 1 form here that submits everything with a button at the top and bottom imo. Will make for a better user experience and a more simplified submit.

Reply:
I tried to make all three firebase calls into promises, like: const namePromise = firebase.auth()...., and then at the end do:  await Promise.All([namePromise, emailPromise, passwordPromise]), but it turned into an error nightmare with some of the promises firing and others not.  I'll need to do a code pair with you or something for this, because it's beyond my understanding.

Reply:
Yeah promise all should work 🤔 Can you post the errors you got and the code snippet?

Reply:
I have three snippets: (1) my code, (2) the errors it threw, (3) the 400 reject error.  I attempted to change my Name and Password, and left my email the same as it was.  The result was that the email and password firebase calls went through successfully, but the Name didn't, and it logged me out.
<img width="780" alt="mycode" src="https://user-images.githubusercontent.com/54561452/74994624-ca157a00-540b-11ea-9b3c-9364f4635e82.png">
![errors](https://user-images.githubusercontent.com/54561452/74994628-cda90100-540b-11ea-9bed-46b4beb0c20c.png)
![400](https://user-images.githubusercontent.com/54561452/74994638-d4d00f00-540b-11ea-8655-b26f80a12d9e.png)

Reply:
Not sure why that would cause an issue, password reset is probably changing the authentication key I bet. Try running everything except the password, then run all on the other two items. Let me know if that works.